src/internal_events/websocket_server.rs

```diff
@@ -0,0 +1,71 @@
+use std::fmt::Debug;
+
+use futures::channel::mpsc::TrySendError;
+use metrics::{counter, gauge};
+use tokio_tungstenite::tungstenite::Message;
+use vector_lib::internal_event::InternalEvent;
+
+use vector_lib::internal_event::{error_stage, error_type};
+
+#[derive(Debug)]
+pub struct WsListenerConnectionEstablished {
+    pub client_count: usize,
+}
+
+impl InternalEvent for WsListenerConnectionEstablished {
+    fn emit(self) {
+        debug!(message = "Websocket client connected.");
+        counter!("connection_established_total").increment(1);
+        gauge!("active_clients").set(self.client_count as f64);
+    }
+
+    fn name(&self) -> Option<&'static str> {
+        Some("WsListenerConnectionEstablished")
+    }
+}
+
+#[derive(Debug)]
+pub struct WsListenerConnectionShutdown {
+    pub client_count: usize,
+}
+
+impl InternalEvent for WsListenerConnectionShutdown {
+    fn emit(self) {
+        warn!(message = "Client connection closed.");
```

Comment:
Is this a warning or an info?

Nit:
```suggestion
        warn!(message = "Client connection closed. Client count: {self.client_count}.");
```

Reply:
Maybe info would make more sense. I took websocket events as an example here and it used warn, but maybe it makes more sense for a client. I guess a lot more disconnections are expected on a server.